src/canister_tests/src/framework.rs

```diff
@@ -513,23 +508,40 @@ pub fn time(env: &StateMachine) -> u64 {
         .as_nanos() as u64
 }
 
-pub fn verify_delegation(user_key: UserKey, signed_delegation: &SignedDelegation, root_key: &[u8]) {
-    // transform delegation into ic typed delegation so that we have access to the signature domain separator
-    // (via as_signed_bytes)
-    let delegation = Delegation::new(
-        signed_delegation.delegation.pubkey.clone().into_vec(),
-        Time::from_nanos_since_unix_epoch(signed_delegation.delegation.expiration),
+pub fn verify_delegation(
+    env: &StateMachine,
+    user_key: UserKey,
+    signed_delegation: &SignedDelegation,
+    root_key: &[u8],
+) {
+    const DOMAIN_SEPARATOR: &[u8] = b"ic-request-auth-delegation";
+
+    // Calculate the delegation hash (which is the signed message)
+    let key_value_pairs = vec![
+        (
+            "pubkey".to_string(),
+            Value::Bytes(signed_delegation.delegation.pubkey.clone().into_vec()),
+        ),
+        (
+            "expiration".to_string(),
+            Value::Number(signed_delegation.delegation.expiration),
+        ),
+    ];
+    // Add signature domain separator,
+    // see https://internetcomputer.org/docs/current/references/ic-interface-spec#authentication for details
+    let mut msg: Vec<u8> = Vec::from([(DOMAIN_SEPARATOR.len() as u8)]);
+    msg.extend_from_slice(DOMAIN_SEPARATOR);
+    msg.extend_from_slice(
+        &ic_representation_independent_hash::representation_independent_hash(&key_value_pairs),
     );
 
-    // this requires imports of internal crypto infrastructure
-    // -> extend state-machine-tests to offer the functionality instead (see L2-739)
-    verify(
-        &delegation.as_signed_bytes(),
-        SignatureBytes(signed_delegation.signature.clone().into_vec()),
-        public_key_bytes_from_der(user_key.as_ref()).unwrap(),
-        &parse_threshold_sig_key_from_der(root_key).unwrap(),
+    env.verify_canister_signature(
+        msg,
+        signed_delegation.signature.clone().into_vec(),
+        user_key.into_vec(),
+        root_key.to_vec(),
     )
-    .expect("signature invalid");
+    .expect("canister signature invalid");
```

Comment:
given the name of the function, maybe use "delegation signature invalid" instead?